test/gtest/ucs/test_datatype.cc

```diff
@@ -1153,22 +1151,30 @@ UCS_TEST_F(test_array, fixed_onstack) {
     test_fixed(&test_array, num_elems);
 }
 
-UCS_PTR_MAP_DEFINE(test_st, 0);
+class test_datatype_ptr_map : public test_datatype {
+public:
+    UCS_PTR_MAP_DEFINE(unsafe_put, 0);
+    UCS_PTR_MAP_DEFINE(safe_put, 1);
+
+    typedef std::map<ucs_ptr_map_key_t, int*> std_map_t;
+    typedef std_map_t::const_iterator std_map_it_t;
+    typedef std::vector<int> std_vec_t;
+    typedef std_vec_t::iterator std_vec_it_t;
+};
 
-UCS_TEST_F(test_datatype, ptr_map) {
-    UCS_PTR_MAP_T(test_st) ptr_map;
+UCS_TEST_F(test_datatype_ptr_map, unsafe_put) {
+    UCS_PTR_MAP_T(unsafe_put) ptr_map;
 
-    const size_t N = 10 * UCS_KBYTE;
-    ucs_ptr_map_key_t ptr_key;
-    std_map_t         std_map;
-    ucs_status_t      status;
+    std_map_t std_map;
+    std_vec_t std_vec(10 * UCS_KBYTE, 0);
```

Comment:
it's confusing to use KBYTE for int elem number (actually array size is 40KB), better to use `10000` or `10*1024` or `1 << 10`